python/ray/autoscaler/_private/local/config.py

```diff
@@ -1,2 +1,98 @@
-def bootstrap_local(config):
+import copy
+from typing import Any
+from typing import Dict
+
+from ray.autoscaler._private.cli_logger import cli_logger
+
+unsupported_field_message = ("The field {} is not supported "
+                             "for on-premise clusters.")
+
+
+def prepare_local(config: Dict[str, Any]) -> Dict[str, Any]:
+    """
+    Prepare local cluster config for ingestion by cluster launcher and
+    autoscaler.
+    """
+    config = copy.deepcopy(config)
+    for field in "head_node", "worker_nodes", "available_node_types":
+        if config.get(field):
+            err_msg = unsupported_field_message.format(field)
+            cli_logger.abort(err_msg)
+    if "coordinator_address" in config["provider"]:
+        config = prepare_coordinator(config)
+    else:
+        config = prepare_manual(config)
+    # Presence of head_node and worker_nodes fields triggers legacy config
+    # processing later in the bootstrap process.
+    config["head_node"] = {"placeholder": {}}
+    config["worker_nodes"] = {"placeholder": {}}
```

Comment:
Why are we doing it in this hacky way? Can we please not do this? didn't we discuss that we should not even need the legacy config processing? Just put everything under the 1 node type and it will work.

Reply:
Once I'm done with the rest of the changes in this PR, I'll try that out -- if it doesn't work out, the options will be likely that we keep it as is or that someone else from our team takes over.

IMO, attempting to use the autoscaler for the simple purpose of bootstrapping a Ray cluster on some private Ray clusters is pretty hacky as it is.

Reply:
Hmm, though the look is pretty hacky -- as a quick fix I'll refactor things to at least avoid having to put these placeholders in.

Reply:
Actually, not entirely sure how to remove that -- will try using a single node type later, but not entirely sure if the resource demand scheduler will be happy with that.

@AmeerHajAli 
My specific concern with using a single node type is that the resource demand scheduler may try to repeatedly infer runtime resources at every autoscaler update, because of this line:
https://github.com/ray-project/ray/blob/e6864523cf6c81e089cbbc41c50260b20ae23b42/python/ray/autoscaler/_private/resource_demand_scheduler.py#L311

Reply:
I am not sure that this should be an issue, can you check it out?

Reply:
Using one node type appears to have worked fine and simplified the code.
Good call!